Unbreak the Android build

bvanassche opened the pull request:
The header file include/sg_pt_linux.h defines the __u64 type and is included even if the <linux/types.h> header is present and defines the __u64 type. Fix this by only defining the __u64 type if it is not defined in <linux/types.h> header.

doug-gilbert commented:
Applied upstream and pushed to this mirror. The Android CI job is "waiting for a runner" ...
Note: version has been bumped to now at pre-release 1.49 .

bvanassche committed:
configure.ac: Fix the getrandom() test

Some systems (older versions of Android) have <sys/random.h> but not
getrandom(). Make sure that presence of getrandom() is detected correctly
on these systems.

Fixes: 9ccb556a21f1 ("sg_dd: setup conditional auto rule for getrandom()")
Signed-off-by: Bart Van Assche <bvanassche@acm.org>

bvanassche committed:
ci: Build against two versions of the Android NDK

Build against a version of the NDK that does not support getrandom() and
against a version that supports getrandom().

Signed-off-by: Bart Van Assche <bvanassche@acm.org>

doug-gilbert commented:
Revision to this PR applied upstream (as a single commit) and pushed here. Thanks.
